### DISCUSSION TITLE
Read a file line by line

### DISCUSSION THREAD

**Discussion by jgomo3** [apache/hop#4705] [Q&A]
How to stream one line at a time from a file?

**Comment by hansva** [comment]
Can you elaborate a bit more? This is hardly a question. The text file input transform will read a file line by line and output the data in rows. But I have no idea what your actual question is.

**Comment by jgomo3** [comment]
I need to report how many vowels each line of a file has.  The file has no more specific format than to be UTF-8 encoded, and lines ending in \n (Unix style).

**Reply by bamaer** [reply]
The Text File Input transform reads and outputs fields from text files line by line by default. If you want to read entire lines instead of separate fields, just set a separator of `#@#@#` or some other nonsense pattern that will never occur in the file, and (manually) specify a single String field (e.g `line`) in the fields tab.
